pkg/controllermanager/controller/plant/plant_control.go

```diff
@@ -0,0 +1,347 @@
+		discoveryClient: make(map[string]*kubernetesclientset.Clientset),
+		healthChecker:   make(map[string]*HealthChecker),
+		plantLister:     plantsLister,
+		secretsLister:   secretLister,
+		recorder:        recorder,
+		config:          config,
+	}
+}
+
+func (c *defaultPlantControl) Reconcile(ctx context.Context, obj *gardencorev1alpha1.Plant, key string) error {
+	var (
+		plant  = obj.DeepCopy()
+		logger = logger.NewFieldLogger(logger.Logger, "plant", plant.Name)
+	)
+
+	if plant.DeletionTimestamp != nil {
+		return c.delete(plant, logger)
+	}
+
+	return c.reconcile(ctx, plant, key, logger)
+}
+
+func (c *defaultPlantControl) conditionThresholdsToProgressingMapping() map[gardencorev1alpha1.ConditionType]time.Duration {
+	out := make(map[gardencorev1alpha1.ConditionType]time.Duration)
+	for _, threshold := range c.config.Controllers.ShootCare.ConditionThresholds {
+		out[gardencorev1alpha1.ConditionType(threshold.Type)] = threshold.Duration.Duration
+	}
+	return out
+}
+
+func (c *defaultPlantControl) updatePlantConditions(plant *gardencorev1alpha1.Plant, conditions ...gardencorev1alpha1.Condition) (*gardencorev1alpha1.Plant, error) {
+	newPlant, err := kutil.TryUpdatePlantConditions(c.k8sGardenClient.GardenCore(), retry.DefaultBackoff, plant.ObjectMeta,
+		func(plant *gardencorev1alpha1.Plant) (*gardencorev1alpha1.Plant, error) {
+			plant.Status.Conditions = conditions
+			return plant, nil
+		})
+
+	return newPlant, err
+}
+
+func (c *defaultPlantControl) reconcile(ctx context.Context, plant *gardencorev1alpha1.Plant, key string, logger logrus.FieldLogger) error {
+	_, err := kutil.TryUpdatePlantStatusWithEqualFunc(c.k8sGardenClient.GardenCore(), retry.DefaultBackoff, plant.ObjectMeta,
+		func(p *gardencorev1alpha1.Plant) (*gardencorev1alpha1.Plant, error) {
+			if finalizers := sets.NewString(p.Finalizers...); !finalizers.Has(FinalizerName) {
+				finalizers.Insert(FinalizerName)
+				p.Finalizers = finalizers.UnsortedList()
+			}
+			return p, nil
+		}, func(cur, updated *gardencorev1alpha1.Plant) bool {
+			return sets.NewString(cur.Finalizers...).Has(FinalizerName)
+		})
+	if err != nil {
+		return err
+	}
+
+	var (
+		newConditions               = helper.MergeConditions(plant.Status.Conditions, helper.InitCondition(gardencorev1alpha1.PlantAPIServerAvailable), helper.InitCondition(gardencorev1alpha1.PlantEveryNodeReady))
+		conditionAPIServerAvailable = newConditions[0]
+		conditionEveryNodeReady     = newConditions[1]
+	)
+
+	defer func() {
+		if _, err := c.updateConditions(plant, conditionAPIServerAvailable, conditionEveryNodeReady); err != nil {
+			logger.Errorf("Failed to update the conditions : %+v", err)
+		}
+	}()
+
+	kubeconfigSecret, err := c.secretsLister.Secrets(plant.Spec.SecretRef.Namespace).Get(plant.Spec.SecretRef.Name)
+	if err != nil {
+		return err
+	}
+	kubeconfig, ok := kubeconfigSecret.Data["kubeconfig"]
+	if !ok {
+		message := "Plant secret needs to contain a valid kubeconfig."
```

Comment:
```suggestion
		message := "Plant secret needs to contain a kubeconfig key."
```

You are not validating here